### ISSUE TITLE
Review handling of time.

### ISSUE THREAD

**Issue by vvorski** [Vorski-Imagineering/share#70]
we have to think through more carefully the handling of time:
- time can never be returned only given
- we keep a track of the balance of time between the two sides. 
- need to think how to display this time balance on the address-book view. 
- input as minutes (15/30/45?), hours, days
### screens to be updated
- friend list - handled in #22 
- friend detail
- create time share
- time share detail
#### on friend detail page need:
- link to their facebook page
- how much time I've given them
- how much time I've received from them
- time balance display

**Comment by konewa** [comment]
@rimbambala design?

**Comment by rimbambala** [comment]
I was thinking of showing time balance in a profile view rather then in address book. In address book we have very limited space for anything else. I will prepare the design in profile view if you like this solution.

**Comment by vvorski** [comment]
@rimbambala  Hmmm. I want to make the idea of time sharing as a key concept of this app. In a lot of ways it is the most imporant part of the whole Share❣ concept...

But maybe you are right putting the whole amount display on this screen won't fit.

How about just an indicator on the person's photo icon that you have some timesharing with them?
Then the details on the profile view as you suggest.

**Comment by vvorski** [comment]
agreed will design, in friends list only show hours... maybe visual display...

**Comment by rimbambala** [comment]
- input as minutes (15/30/45?), hours, days
  as we agreed it would be basically hours and optional minutes.

My proposition for UI would be:

web app - simple combobox

Android devices - second dialog form the screen - http://developer.android.com/design/media/dialogs_examples.png

iOS devices - Picker from the screen - https://developer.apple.com/library/ios/documentation/userexperience/conceptual/MobileHIG/Art/picker_2x.png

Hours range 1 to 24, minutes range 0, 15, 30, 45

**Comment by vvorski** [comment]
How will the system picker look in our design? Is there an option for a skinned control?

**Comment by konewa** [comment]
@rafallo  ?

**Comment by rafallo** [comment]
On iOS we can show native date picker and now it works that way. 
Android devices are not able to use native date picker in browser so we have to use something Javascript based.

**Comment by rimbambala** [comment]
Maybe in case of android we can use a simple popup/modal just like the one used for filtering or language change

**Comment by vvorski** [comment]
- on listing page only show hours
- on detail page show hours and minutes.

**Comment by rimbambala** [comment]
Listing page means address book / friend list?

**Comment by vvorski** [comment]
@rimbambala listing page is friends list - handled in #22 so that leaves:
- friend detail
- create time share
- time share detail
- timeline (same as timeshare detail or different?)

**Comment by vvorski** [comment]
### on friend detail page need:
- link to their facebook page
- how much time I've given them
- how much time I've received from them
- time balance display

**Comment by rimbambala** [comment]
Create Time Share
![share-newtimeshare-step2-12](https://cloud.githubusercontent.com/assets/6513797/2602839/1ce8a3b8-bb26-11e3-90c9-697186f4aca2.jpg)

Timeline time share view (no photo; if photo is present here timeshare will look just as other ones - the only difference is that we are showing time given/received)
![share-06f](https://cloud.githubusercontent.com/assets/6513797/2602870/94b6935a-bb26-11e3-80e3-a384d191e962.jpg)

Time share detail would be just like detailed thing/promise share

**Comment by vvorski** [comment]
Ok, except need to also be able to input text with time - same as others photo or text or both...

Note new suggested create dialog layout in #65.

**Comment by vvorski** [comment]
- also what will the time selection look like when clicked. are we going to use a custom selector?

**Comment by rimbambala** [comment]
![share-newtimeshare-step2-13](https://cloud.githubusercontent.com/assets/6513797/2641856/c80aa232-beff-11e3-9c1c-ed9ff1d07565.jpg)
New share with "add text" feature 

In case of time selection my proposal was:

My proposition for UI would be:

web app - simple combobox

Android devices - second dialog form the screen - http://developer.android.com/design/media/dialogs_examples.png

iOS devices - Picker from the screen - https://developer.apple.com/library/ios/documentation/userexperience/conceptual/MobileHIG/Art/picker_2x.png

Everything depends on development possibility. Correct me if I'm wrong, I think @rafallo said that it might be impossible for picker to be incorporated as far as Android devices are concerned. In that case we can use the first dialog from this link:
http://developer.android.com/design/media/dialogs_examples.png

We can than style it as any other modal in the application. E.g. 
![share-06-language](https://cloud.githubusercontent.com/assets/6513797/2642005/cf90f5cc-bf01-11e3-8558-2f4f6ce167c6.jpg)

Every modal should be design in similar theme.

**Comment by rimbambala** [comment]
![share-07-profileview](https://cloud.githubusercontent.com/assets/6513797/2643611/1b95b900-bf1e-11e3-9d05-08897be0c162.jpg)

**Comment by vvorski** [comment]
Detail page we should simplify and use that icon from the listing page (#22) also let's see if we can add to the icon indicator for whether we received or gave, something based on hands perhaps?
- hands up I am in plus - I am giving
- hands down/around bottom - I am receiving

![person-profile](https://cloud.githubusercontent.com/assets/1789426/2650811/015836e2-bf7a-11e3-81f7-b8f769d9564b.jpg)

**Comment by vvorski** [comment]
For time input wouldn't we be better off using our own controls? The system controls look very different from our UI...

**Comment by rimbambala** [comment]
![share-08-profileview](https://cloud.githubusercontent.com/assets/6513797/2669084/c0d447dc-c0be-11e3-8fe3-a3491a4ae9c1.jpg)

**Comment by vvorski** [comment]
What about:
- indication of giving/receiving on the icon?
-

**Comment by rimbambala** [comment]
![share-08-profileview](https://cloud.githubusercontent.com/assets/6513797/2696604/a5202906-c3e4-11e3-8813-e2d9cee50818.jpg)

**Comment by vvorski** [comment]
@konewa  ok let's just go with this for now... need to implement and launch...
